### DISCUSSION TITLE
How to apply bm25s to languages such as Chinese?

### DISCUSSION THREAD

**Discussion by AlanLu0808** [xhluca/bm25s#36] [Q&A] [answered]
I think the bm25s library is great and very efficient. I would like to use it in my project. 

But how to apply bm25s to languages such as Chinese? Can you provide some examples?

**Comment by bm777** [comment, marked as answer]
It works for Chinese language too. just use a Chinese tokenizer and stemmer.

**Reply by IeohMingChan** [reply]
How to use chinese tokenizer and stemmer, can you provide any example? thanks

**Reply by bm777** [reply]
Here you go

```python
import bm25s

# Create your corpus here
corpus = [
    "貓是貓科動物，喜歡發出咕嚕聲",
    "狗是人類最好的朋友，喜歡玩耍"
]

# Tokenize the corpus
corpus_tokens = bm25s.tokenize(corpus, stopwords="zh")

```

**Reply by IeohMingChan** [reply]
After testing, it was found that due to the vocabulary in Chinese sentences is not connected by spaces and punctuation marks like in English, the BM25S tokenizer itself cannot separate the vocabulary in Chinese sentences. Therefore, Chinese sentences need to be pre splited, such as using jieba, then connect the separated vocabulary and input into the BM25S tokenizer.

**Reply by bm777** [reply]
btw BM25 tokenizer is directly coming from NLTK library.
https://github.com/xhluca/bm25s/blob/8b36bdd69dcbc0405585296ca18f20bd93e197b7/bm25s/tokenization.py#L50-L84

**Reply by IeohMingChan** [reply]
Using Chinese stop words does not split Chinese sentences. Sentences are split based on token_pattern, and the regular expressions used by token_pattern cannot split Chinese sentences. I think Chinese stop words need to be used after splitting Chinese sentences.

**Reply by bm777** [reply]
@xhluca what do you think?

**Reply by xhluca** [reply]
You can use the stopwords separately for now. In v0.2 you will have a tokenizer class that can take a splitter function as input, so you can pass jieba function to it

**Comment by bm777** [comment]
@AlanLu0808 have a look here: #33

**Reply by xansar** [reply]
> @AlanLu0808看看这里：#33

The tokenize function needs a 'token_pattern' to split words, which may not work for chinese. Using a split func as the input parameter may be better.
